### PR TITLE
Eliminate expressionsem import from attrib.d.

### PR DESCRIPTION
**Pull request by veelo**
In pursuit of https://github.com/orgs/dlang/projects/41.

### CODE REVIEW

**Comment by veelo** [comment]
There is trouble with C++ interop: https://github.com/dlang/dmd/actions/runs/8513938331/job/23318627376?pr=16354#step:19:169

How do I resolve this? @RazvanN7

File: compiler/src/dmd/attribsem.d [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import dmd.attrib;
+import dmd.dscope;
+import dmd.dsymbol;
+import dmd.expression;
+import dmd.expressionsem;
+import dmd.location;
+import dmd.root.array; // for each
+
+
+Expressions* getAttributes(UserAttributeDeclaration a)
```
Comment (ibuclaw) [inline compiler/src/dmd/attribsem.d:37]: Used by gdc (and likely ldc too), so needs exporting to C++.

See for example: https://github.com/dlang/dmd/pull/16288

1. Move `getAttributes` in `dmd/attrib.h` to `namespace dmd`
2. Add C++->D wrapper to `cxxfrontend.d`

Comment (veelo) [reply compiler/src/dmd/attribsem.d:37]: Thank you.

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
Remaining failures are code coverage upload failures.

**Pull request merged**